### ISSUE TITLE
Simplify form integral id handling

### ISSUE THREAD

**Issue by garth-wells** [FEniCS/ffcx#103]
`ufc_form` integrals are unnecessarily complex, with, for example:
```
/// Create a new cell integral on sub domain subdomain_id 
ufc_cell_integral* (*create_cell_integral)(int subdomain_id);

/// Create a new cell integral on everywhere else
ufc_cell_integral* (*create_default_cell_integral)(void);

/// Upper bound on subdomain ids for cell integrals
int max_cell_subdomain_id;
```
where `max_cell_subdomain_id` could possible be very large.

More helpful would be: 
```
/// Create a new cell integral on sub domain subdomain_id 
ufc_cell_integral* (*create_cell_integral)(int subdomain_id);

/// Number of subdomain ids for cell integrals, and array of ids
void ufc_cell_integral_ids (**ids, int& num_ids);
```
where the default integral could be `-1`.

**Comment by chrisrichardson** [comment]
Looks good. I like the idea that the default integral should be just another integral, and not require a special function, and this would be a way to achieve it.

**Comment by chrisrichardson** [comment]
Work in progress in branch chris/simplify-integrals

**Comment by chrisrichardson** [comment]
Oops just noticed you've already done it...

**Comment by garth-wells** [comment]
@chrisrichardson I haven't really done it - just some debug to figure what was happening in UFL.

**Comment by chrisrichardson** [comment]
OK. sorry that was the dolfinx issue... getting confused. Hopefully this will help.

**Comment by chrisrichardson** [comment]
Resolved with commit 53dc20094d017017537fc3010a78d29d1652effc
